File: Jzzhu and Children/jzzhu_and_children.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

int Jzzhu_and_Children()
{ ll total_students,chocolates_per_day;
  cin>>total_students>>chocolates_per_day;
  ll chocolates_required[total_students];
  ll in_step[total_students],max1 = -1 ;
  
  for(int i = 0;i < total_students;i++)
  {   cin>>chocolates_required[i];
      float times = float(chocolates_required[i]) / chocolates_per_day;
      in_step[i] = ceil(times);
      max1 = max(max1,in_step[i]);
      
  }
  ll length = total_students - 1 ;
  
 

  while (length >= 0)
  {  if(in_step[length] == max1)
       return (length + 1) ;
     length--; 
  }
   
   
}


int main() {
	int remained_test_cases = 1 ;
// 	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   cout<<Jzzhu_and_Children();
	    remained_test_cases--;
	}
	return 0;
}

File: Jzzhu and Children/jzzhu_and_children.py
import math
def Jzzhu_and_Children():
    total_childrens,chocolate_per_step = map(int,input().split())
    chocolate_need = list(map(int,input().split()))
    in_turns = []
    
    for choc in range(total_childrens):
        in_turns.append(math.ceil(chocolate_need[choc]/chocolate_per_step))
    max1 = max(in_turns)
    # print(in_turns)
    
    length = total_childrens - 1 
    while length >= 0:
        if in_turns[length] == max1:
            return length+1 
        length -= 1     
    
    
remained_test_cases = 1 
# remained_test_cases = int(input())
while remained_test_cases > 0:
    print(Jzzhu_and_Children())
    remained_test_cases -= 1 